presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java

```diff
@@ -0,0 +1,216 @@
+
+    private final HttpClient httpClient;
+    private final URI druidCoordinator;
+    private final URI druidBroker;
+
+    @Inject
+    public DruidClient(DruidConfig config, @ForDruidClient HttpClient httpClient)
+    {
+        requireNonNull(config, "config is null");
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+        this.druidCoordinator = URI.create(config.getDruidCoordinatorUrl());
+        this.druidBroker = URI.create(config.getDruidBrokerUrl());
+    }
+
+    public URI getDruidBroker()
+    {
+        return druidBroker;
+    }
+
+    public List<String> getTables()
+    {
+        return httpClient.execute(prepareQuery(LIST_TABLE_QUERY), createJsonResponseHandler(LIST_TABLE_NAME_CODEC)).stream()
+                .map(DruidTableInfo::getTableName)
+                .collect(toImmutableList());
+    }
+
+    public List<DruidColumnInfo> getColumnDataType(String tableName)
+    {
+        return httpClient.execute(prepareQuery(format(GET_COLUMN_TEMPLATE, tableName)), createJsonResponseHandler(LIST_COLUMN_INFO_CODEC));
+    }
+
+    public List<String> getAllDataSegmentId(String tableName)
+    {
+        return getDataSegmentIdInDomain(tableName, Domain.all(TIMESTAMP));
+    }
+
+    public List<String> getDataSegmentIdInDomain(String tableName, Domain domain)
+    {
+        return httpClient.execute(prepareQuery(format(GET_SEGMENTS_ID_TEMPLATE, tableName, toPredicate(domain))), createJsonResponseHandler(LIST_SEGMENT_ID_CODEC)).stream()
+                .map(wrapper -> wrapper.getSegmentId())
+                .collect(toImmutableList());
+    }
+
+    public DruidSegmentInfo getSingleSegmentInfo(String dataSource, String segmentId)
+    {
+        URI uri = uriBuilderFrom(druidCoordinator)
+                .replacePath(METADATA_PATH)
+                .appendPath(format("datasources/%s/segments/%s", dataSource, segmentId))
+                .build();
+        Request request = setContentTypeHeaders(prepareGet())
+                .setUri(uri)
+                .build();
+
+        return httpClient.execute(request, createJsonResponseHandler(SEGMENT_INFO_CODEC));
+    }
+
+    private static Request.Builder setContentTypeHeaders(Request.Builder requestBuilder)
+    {
+        return requestBuilder
+                .setHeader(CONTENT_TYPE, JSON_UTF_8.toString())
+                .setHeader(ACCEPT, JSON_UTF_8.toString());
+    }
+
+    private static byte[] createRequestBody(String query)
+    {
+        return format("{\"query\":\"%s\"}\n", query).getBytes();
+    }
+
+    private Request prepareQuery(String query)
+    {
+        HttpUriBuilder uriBuilder = uriBuilderFrom(druidBroker).replacePath(SQL_ENDPOINT);
+
+        return setContentTypeHeaders(preparePost())
+                .setUri(uriBuilder.build())
+                .setBodyGenerator(createStaticBodyGenerator(createRequestBody(query)))
+                .build();
+    }
+
+    private String toPredicate(Domain domain)
```

Comment:
Do you want to use the Pushdown framework to push the filters down instead of Tuple Domains ?

Reply:
yep, that's a following step